// @data-ui/radial-chart/build/chart/RadialChart.js
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports['default'] = RadialChart;

var _group = require('@vx/group');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var propTypes = {
  ariaLabel: _propTypes2['default'].string.isRequired,
  children: _propTypes2['default'].node.isRequired,
  width: _propTypes2['default'].number.isRequired,
  height: _propTypes2['default'].number.isRequired,
  margin: _propTypes2['default'].shape({
    top: _propTypes2['default'].number,
    right: _propTypes2['default'].number,
    bottom: _propTypes2['default'].number,
    left: _propTypes2['default'].number
  })
};

var defaultProps = {
  margin: {
    top: 10,
    left: 10,
    right: 10,
    bottom: 10
  }
};

function RadialChart(_ref) {
  var ariaLabel = _ref.ariaLabel,
      children = _ref.children,
      width = _ref.width,
      height = _ref.height,
      margin = _ref.margin;

  var completeMargin = Object.assign({}, defaultProps.margin, margin);
  var innerWidth = width - completeMargin.left - completeMargin.right;
  var innerHeight = height - completeMargin.top - completeMargin.bottom;
  var radius = Math.min(innerWidth, innerHeight) / 2;
  if (innerWidth < 10 || innerHeight < 10) return null;

  return _react2['default'].createElement(
    'svg',
    {
      'aria-label': ariaLabel,
      role: 'img',
      width: width,
      height: height
    },
    _react2['default'].createElement(
      _group.Group,
      {
        top: height / 2 - completeMargin.top,
        left: width / 2 + margin.left
      },
      _react2['default'].Children.map(children, function (Child) {
        return _react2['default'].cloneElement(Child, { radius: radius });
      })
    )
  );
}

RadialChart.propTypes = propTypes;
RadialChart.defaultProps = defaultProps;
RadialChart.displayName = 'RadialChart';
