// @data-ui/radial-chart/build/label/ArcLabel.js
Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports['default'] = ArcLabel;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _svgLabel = require('@data-ui/theme/build/svgLabel');

var _additionalProps = require('../util/additionalProps');

var _additionalProps2 = _interopRequireDefault(_additionalProps);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  x: _propTypes2['default'].number,
  y: _propTypes2['default'].number,
  arc: _propTypes2['default'].object, // if passed, will be passed to any restprops
  children: _propTypes2['default'].node
};

var defaultProps = {
  x: 0,
  y: 0,
  paintOrder: 'stroke',
  arc: null,
  children: null
};

function ArcLabel(_ref) {
  var x = _ref.x,
      y = _ref.y,
      children = _ref.children,
      arc = _ref.arc,
      rest = _objectWithoutProperties(_ref, ['x', 'y', 'children', 'arc']);

  return _react2['default'].createElement(
    'text',
    _extends({
      x: x,
      y: y
    }, _svgLabel.baseLabel, (0, _additionalProps2['default'])(rest, arc)),
    children
  );
}

ArcLabel.propTypes = propTypes;
ArcLabel.defaultProps = defaultProps;
