// @data-ui/forms/build/Button.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _aphrodite = require('aphrodite');

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var unit = 8;

var styles = _aphrodite.StyleSheet.create({
  container: {
    cursor: 'pointer',
    transition: 'background 0.3s, border-color 0.3s',
    border: 0,
    borderRadius: 1,
    background: '#fff',
    padding: 1 * unit,
    color: '#484848',
    outline: 'none',

    ':hover': {
      background: '#eaeaea'
    },

    ':active': {
      background: '#ddd'
    }
  },

  border: {
    boxShadow: 'inset 0 0 0 1px #ddd'
  },

  text: {
    pointerEvents: 'none'
  },

  text_sizeSmall: {
    fontSize: 12
  },

  text_sizeRegular: {
    fontSize: 14
  },

  container_sizeSmall: {
    padding: 1 * unit
  },

  container_sizeRegular: {
    padding: 1.5 * unit + 'px ' + 2 * unit + 'px'
  },

  container_block: {
    width: '100%'
  },

  container_notBlock: {
    width: 'auto'
  },

  text_wrap: {
    whiteSpace: 'normal'
  },

  text_noWrap: {
    whiteSpace: 'nowrap'
  },

  rounded: {
    borderRadius: unit / 2
  },

  round: {
    width: 4 * unit,
    height: 4 * unit,
    lineHeight: 4 * unit + 'px',
    borderRadius: '50%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 0
  },

  round_small: {
    width: 3 * unit,
    height: 3 * unit,
    lineHeight: 3 * unit + 'px',
    borderRadius: '50%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 0
  },

  disabled: {
    opacity: 0.5,
    cursor: 'not-allowed'
  }
});

var propTypes = {
  block: _propTypes2.default.bool,
  children: _propTypes2.default.node,
  onClick: _propTypes2.default.func,
  disabled: _propTypes2.default.bool,
  small: _propTypes2.default.bool,
  wrapText: _propTypes2.default.bool,
  rounded: _propTypes2.default.bool,
  round: _propTypes2.default.bool,
  noBorder: _propTypes2.default.bool
};

var defaultProps = {
  block: false,
  children: null,
  onClick: function onClick() {},
  disabled: false,
  small: false,
  wrapText: false,
  rounded: false,
  round: false,
  noBorder: false
};

function Button(_ref) {
  var block = _ref.block,
      children = _ref.children,
      onClick = _ref.onClick,
      disabled = _ref.disabled,
      small = _ref.small,
      round = _ref.round,
      rounded = _ref.rounded,
      noBorder = _ref.noBorder,
      wrapText = _ref.wrapText;

  return _react2.default.createElement(
    'button',
    {
      'aria-disabled': disabled,
      disabled: disabled,
      onClick: onClick,
      className: (0, _aphrodite.css)(styles.container, !noBorder && styles.border, block ? styles.container_block : styles.container_notBlock, small ? styles.container_sizeSmall : styles.container_sizeRegular, rounded && styles.rounded, round && small ? styles.round_small : round && styles.round, disabled && styles.disabled)
    },
    _react2.default.createElement(
      'span',
      {
        className: (0, _aphrodite.css)(styles.text, small ? styles.text_sizeSmall : styles.text_sizeRegular, wrapText ? styles.text_wrap : styles.text_noWrap)
      },
      children
    )
  );
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

exports.default = Button;
